controllers/argocd/deployment.go

```diff
@@ -332,6 +333,11 @@ func (r *ReconcileArgoCD) reconcileDeployments(cr *argoprojv1a1.ArgoCD) error {
 
 // reconcileDexDeployment will ensure the Deployment resource is present for the ArgoCD Dex component.
 func (r *ReconcileArgoCD) reconcileDexDeployment(cr *argoprojv1a1.ArgoCD) error {
+	if cr.Spec.SSO != nil {
+		err := e.New("duplicated SSO configuration")
+		log.Error(err, fmt.Sprintf("Only can install one SSO provider, please uninstall keycloak for ArgoCD %s in namespace %s",
```

Comment:
Please see https://github.com/argoproj-labs/argocd-operator/pull/442/files#r714119303

Reply:
Thanks for the review, this is updated as you suggested.